torchtune/modules/transformer.py

```diff
@@ -293,6 +291,8 @@ class TransformerDecoder(nn.Module):
             before final MLP.
         output (nn.Linear): Callable that applies a linear transformation to the output of
             the decoder.
+        num_layers (Optional[int]): Number of Transformer Decoder layers, only define when
+            layers is not a list.
```

Comment:
This was in the TransformerLayer refactor. I left it in to support both because otherwise the builders get a lot less clean. But We could update that in the future.